src/client/dfuse/ops/symlink.c

```diff
@@ -39,8 +39,8 @@ dfuse_cb_symlink(fuse_req_t req, const char *link,
 
 	DFUSE_TRA_UP(ie, parent, "inode");
 
-	rc = dfs_open(parent->ie_dfs->dfs_ns, parent->ie_obj, name, S_IFLNK,
-		      O_CREAT, 0, 0, link, &ie->ie_obj);
+	rc = dfs_open(parent->ie_dfs->dfs_ns, parent->ie_obj, name,
+		      S_IFLNK | 0777, O_CREAT, 0, 0, link, &ie->ie_obj);
```

Comment:
Instead of setting the symlink permissions to 777 at the `dfuse` level, do you think it would be better to set it at the `DFS` level? Within mpifileutils, we call the `dfs_*` functions directly (bypassing `dfuse`), so I'm thinking that if we can put this at the `DFS` level then no matter which library calls `DFS`, it will still work the same, including any future applications built on top of `DFS` directly. And then, similar to POSIX, `DFS` could disallow `chmod` on a symlink's permissions, dereferencing symlinks by default.
https://github.com/daos-stack/daos/blob/c058beb1612e6d6abaf7164f2dcd6f492165415e/src/client/dfs/dfs.c#L834
We could do something like:
`entry.mode = sym->mode | S_IRWXO | S_IRWXU | S_IRWXG;`

Reply:
Yes, that's a good idea.  Will refresh.

Reply:
Actually, it does seem that `dfs_chmod` dereferences symlinks, so I don't think we need to worry about `dfs_chmod` disallowing setting symlink permissions, but I don wonder if `dfs_osetattr` should disallow setting symlinks `mode`/permissions? (Or maybe this is beyond the scope of this PR)

Reply:
Also, it seems that `dfs_osetattr` doesn't handle all edges cases of trying to set different modes, such as changing the filetype. E.g. if you change from symlink to directory, the symlink value needs to be freed, or if you change from regular file to directory, the array probably needs to be punched.
So, I'm thinking I'll write this up in a separate ticket

Reply:
Beyond the scope of this PR.

We shouldn't allow osetattr() to change the file type.  A separate ticket would be best.

Reply:
I'll make the change to the default mode of symlinks however.

Reply:
@ashleypittman Were you planning to handle the symlink permissions on this PR, or separately for [DAOS-5821](https://jira.hpdd.intel.com/browse/DAOS-5821)?

Reply:
I put it in this PR after you mentioned it but before I realised it was a seperate ticket.  I can update the description of this one to include that ticket number but I don't see any reason to land it separately.

Reply:
I don't think it necessarily needs to be landed separately, but I do think it would be useful for the default mode of 777 to be done in DFS instead of DFUSE, so that anything built on top of DFS behaves the same way. (I think this discussion just got lost in the noise)

Reply:
Ah, I thought it was.

Let me change this then.